networking/src/Network/Transport/ConnectionBuffers.hs

```diff
@@ -1,290 +0,0 @@
-                      }
-                in  return (st', buffer)
-          writeBuffer buffer Closed
-
-        Received connid bytes -> do
-          buffer <- withSharedAtomic qdiscState $ \st ->
-            case M.lookup connid (qdiscBuffers st) of
-              Nothing          -> internalError UnknownConnectionReceived
-              Just (_, buffer) -> return buffer
-          -- Will block if the buffer is full, ultimately blocking the thread
-          -- which is reading from the socket.
-          writeBufferRespectingChunkSize buffer (BL.fromChunks bytes)
-
-        EndPointClosed ->
-          let event = LocalEndPointClosed
-          in  writeBuffer endPointBuffer event
-
-        ErrorEvent (TransportError (EventError EventEndPointFailed) _) ->
-          let event = LocalEndPointFailed
-          in  writeBuffer endPointBuffer event
-
-        ErrorEvent (TransportError (EventError EventTransportFailed) _) ->
-          let event = LocalTransportFailed
-          in  writeBuffer endPointBuffer event
-
-        -- Must find all connections for that address and write 'Lost' to their
-        -- buffers.
-        ErrorEvent (TransportError (EventError (EventConnectionLost addr)) _) -> do
-          buffers <- modifySharedAtomic qdiscState $ \st ->
-            case M.lookup addr (qdiscConnections st) of
-              -- TBD: will network-transport report all of the connections to
-              -- this peer as closed, before giving the connection lost
-              -- message? I think so but I'm not sure.
-              Nothing -> return (st, [])
-              -- Get the buffers for all of the ConnectionIds and remove them
-              -- from the state.
-              -- connids is guaranteed non-empty set but GHC doesn't know that.
-              Just connids ->
-                let combine connid (buffers, deletedBuffers) =
-                      case M.updateLookupWithKey (const (const Nothing)) connid buffers of
-                        -- updateLookupWithKey returns the deleted value if it
-                        -- was deleted.
-                        (Just (_, buffer), buffers') -> (buffers', Just buffer : deletedBuffers)
-                        _                            -> (buffers, Nothing : deletedBuffers)
-                    (buffers, deletedBuffers) = foldr combine (qdiscBuffers st, []) (S.toList connids)
-                    st' = st {
-                        qdiscBuffers = buffers
-                      , qdiscConnections = M.delete addr (qdiscConnections st)
-                      }
-                in  return (st', deletedBuffers)
-          forM_ buffers $ maybe (internalError InconsistentConnectionState) (flip writeBuffer Lost)
-
-  return $ QDisc enqueue dequeue
-
-data InternalError =
-    InconsistentConnectionState
-  | UnknownConnectionClosed
-  | UnknownConnectionLost
-  | UnknownConnectionReceived
-  | DuplicateConnection
-  deriving (Typeable, Show)
-
-instance Exception InternalError
```

Comment:
It would be great to have @avieth confirm deletion of this module. It isn't included in the respective cabal file and seems fairly bitrotten.

Reply:
Yeah this was just a sketch and shouldn't have been committed at all.